### PR TITLE
feat(todo|allure): Add method `todo` for planning on writing tests

### PR DESCRIPTION
**Pull request by Vorobeyko**
## Motivation/Description of the PR

Applicable helpers:

- [ ] Webdriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe

- Description of this PR, which problem it solves
- A link to the corresponding issue (if applicable).

## Type of change

- [ ] Breaking changes
- [x] New functionality
- [ ] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [x] Tests have been added
- [x] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: lib/ui.js [hunk 1]
```diff
@@ -141,10 +141,48 @@ module.exports = function (suite) {
...
+
+        return context.Scenario(title, {
+          skipInfo: context.Scenario.skipInfo,
+        });
+      }
+
       return context.Scenario(title, {});
     };
 
+    context.Scenario.todo = function (title, opts = {}, fn) {
```
[the remark this answers is not in the document]
Comment (Vorobeyko) [reply lib/ui.js]: Of course!)

**Comment by Vorobeyko** [comment]
Need add docs to https://codecept.io/hooks/#api

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/data/sandbox/configs/todo/first_test.js  3
- test/runner/todo_test.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4723101&bid=15768639)

File: lib/ui.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vorobeyko) [reply lib/ui.js]: @PeterNgTr Tests added

File: lib/plugin/allure.js [hunk 2]
```diff
@@ -74,22 +74,40 @@ const defaultConfig = {
...
 
-  this.addAttachment = (name, buffer, type) => {
+  reporter.pendingCase = function (testName, timestamp, opts = {}) {
+    reporter.startCase(testName, timestamp);
+
+    if (opts.description) plugin.setDescription(opts.description);
+    if (opts.severity) plugin.severity(opts.severity);
+    if (opts.severity) plugin.addLabel('tag', opts.severity);
+
+    reporter.endCase('pending', { message: opts.message || 'Test ignored' }, timestamp);
```
Comment (PeterNgTr) [inline lib/plugin/allure.js]: what do you think if we use `Pending Test` to make it more consistent :)

Comment (Vorobeyko) [reply lib/plugin/allure.js]: 'Test ignored' - is default message from allure-js-common)

Comment (PeterNgTr) [reply lib/plugin/allure.js]: Didn't know that 😺

**Comment by PeterNgTr** [comment]
@Vorobeyko just curious how this will be displayed when using `dry-run`. Other than that I have nothing more.

**Comment by Vorobeyko** [comment]
> @Vorobeyko just curious how this will be displayed when using `dry-run`. Other than that I have nothing more.

Like skipped test)

**Comment by DavertMik** [comment]
Nice. Let's have this!

**Pull request merged**